rayon-core/src/scope/mod.rs

```diff
@@ -37,21 +38,31 @@ pub struct ScopeFifo<'scope> {
     fifos: Vec<JobFifo>,
 }
 
-struct ScopeBase<'scope> {
-    /// thread where `scope()` was executed (note that individual jobs
-    /// may be executing on different worker threads, though they
-    /// should always be within the same pool of threads)
-    owner_thread_index: usize,
+enum ScopeLatch {
+    /// A latch for scopes created on a rayon thread which will participate in work-
+    /// stealing while it waits for completion. This thread is not necessarily part
+    /// of the same registry as the scope itself!
+    Stealing {
+        latch: CountLatch,
+        registry: Arc<Registry>,
```

Comment:
I'm curious what @cuviper thinks: I'd be inclined not to store this in a field but instead to pass an `&Arc<Registry>` as an argument to `ScopeLatch::set`. It avoids an extra atomic increment on every `scope`.

Reply:
What about the cross-registry case? If we call `ThreadPool::in_place_scope()` on registry B from a worker thread of registry A, when a job completes on a worker thread of registry B, we would need to call `ScopeLatch::set` with a reference to registry A but we won't be able to find one.

Reply:
I must not have my head in the right space; I don't quite follow. Maybe you can add a nice comment into the code detailing how that scenario plays out and why this field is necessary?

Reply:
I added a comment but I'm not sure I've made it any clearer than what I already said.

When a job completes in registry B we only have a reference to registry B, but we may need to wake the thread in registry A, for which we need a reference to registry A.

Reply:
Thanks. That helps. I'll try to make some time to read it over. I'm still contemplating what happened in the old code for this logic -- I guess the point is that this case didn't arise, because we always executed the scope's closure over within the target registry, so there was no cross-registry case? (Or at least, it was handled by the logic that moved over into the other registry?)

Reply:
> I guess the point is that this case didn't arise, because we always executed the scope's closure over within the target registry, so there was no cross-registry case? (Or at least, it was handled by the logic that moved over into the other registry?)

Correct -- it would have ended up calling `scope` -> `in_worker` -> `in_worker_cross` to handle that case.